From e9b4338aac031f2a1ec5dfd72762dfdbc81f9bd3 Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Wed, 10 Feb 2021 14:44:46 +0200
Subject: [PATCH 07/15] phc2sys: reduce sync delay after enabling extts

Now phc2sys waiting 3 sec before starting sync, this delay is not needed.

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 phc2sys.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/phc2sys.c b/phc2sys.c
index 8d17aa4..f75ed2e 100644
--- a/phc2sys.c
+++ b/phc2sys.c
@@ -1084,7 +1084,7 @@ static void autocfg_extts_clock_settime(struct node *node,
 		tdst2.tv_sec, tdst2.tv_nsec);
 }
 
-#define CPTS_EXTTS_DISCARDS_MAX  3
+#define CPTS_EXTTS_DISCARDS_MAX  0
 #define OTHER_EXTTS_DISCARDS_MAX (CPTS_EXTTS_DISCARDS_MAX + 7)
 #define INIT_EXTTS_DIFF_MAX      500000000
 
@@ -1338,8 +1338,10 @@ static int do_autocfg_extts_loop(struct node *node, int subscriptions)
 				continue;
 
 			/* skip N extts to have PPS settle */
-			if (clock->extts_discard--)
+			if (clock->extts_discard) {
+				clock->extts_discard--;
 				continue;
+			}
 
 			update_clock(node, clock, extts_offset, extts_ts, -1);
 		}
-- 
2.17.1

